cmd/scollector/collectors/httpunit.go

```diff
@@ -0,0 +1,79 @@
+package collectors
+
+import (
+	"fmt"
+
+	"bosun.org/_third_party/github.com/BurntSushi/toml"
+	"bosun.org/_third_party/github.com/StackExchange/httpunit"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+)
+
+func HTTPUnitFile(filename string) error {
+	var plans httpunit.Plans
+	if _, err := toml.DecodeFile(filename, &plans); err != nil {
+		return err
+	}
+	HTTPUnitPlans(filename, &plans)
+	return nil
+}
+
+func HTTPUnitPlans(name string, plans *httpunit.Plans) {
+	collectors = append(collectors, &IntervalCollector{
+		F: func() (opentsdb.MultiDataPoint, error) {
+			return cHTTPUnit(plans)
+		},
+		name: fmt.Sprintf("c_httpunit_%s", name),
+	})
+}
+
+func cHTTPUnit(plans *httpunit.Plans) (opentsdb.MultiDataPoint, error) {
+	ch, _, err := plans.Test("", false)
```

Comment:
general api design question: What happens with errors that happen later in the check process? Do they get propagated through the result channel, or just swallowed?